### DISCUSSION TITLE
Using local API

### DISCUSSION THREAD

**Discussion by JWdeBokx** [ekutner/home-connect-hass#388] [Ideas]
Hello,

I'm using the integration just shortly, and it's great.
Saw some discussion where the local api is mentioned

On another site I found a way to use the local api to send mqtt messages.
But I'm not a python developer, I was a windows developer.

Do you still have time for this addon to improve things?
I assume you are doing this in your spare time like most addon builders....

**Comment by ekutner** [comment]
See the discussion here #271 
If you know of a way to do it let me know but reverse engineering the local API is beyond what I have the time to do.

**Reply by JWdeBokx** [reply]
Thank you for your reply.

I did found some sites with also some code.
But python is nothing for me.
I can read it globally, but thats it

the sites I found:
[https://trmm.net/homeconnect/](url)
The first half is nice to know, but search for How do I use it? that's where it actually starts.
Some things I have tried, and actually gives the described result.
But then... I don't know
The code is found here: [https://github.com/osresearch/hcpy](url)

Another site is this: [https://github.com/eifel-tech/ioBroker.cloudless-homeconnect](url)
It's not for homeassistant, but maybe you can find the api details in here?

I'm glad with al the time you have available for this.
If this does not help, then we just need to work with the cloud solution. It's does work great.

**Comment by ekutner** [comment]
The first one is the one mentioned in the above discussion, the second one is a port of the first from Python to JS. Either way, this is very far from a complete project that I can just take and use in the integration and will require quite a lot of reverse engineering to port to something useful. Since things are mostly working well enough with the Home Connect cloud, and since it will be a very big effort to support this local mode, it will require much more than I can put into this project at this time.

**Reply by JWdeBokx** [reply]
Ah oke,
I had no idea how far the code was.
Thank you for looking into it.
Indeed most of tthe time the cloud version works.
It's just some functions a use a lot are not available of my neff oven.
In the app with the local api it is.

You did already a nice job, I can't do it!
So thank for your time in this question.
